### PR TITLE
tree-sitter: use new highlight groups

### PR DESCRIPTION
**Pull request by leana8959**
Use new tree-sitter highlight groups

Highlight groups can be found [here](https://github.com/nvim-treesitter/nvim-treesitter/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by leana8959** [comment]
The commit history is borked, I'll clean everything up later

File: lua/curry/theme.lua [hunk 1]
```diff
@@ -99,32 +99,100 @@ local highlight_groups = {
...
 
-	["@text.literal.block.markdown"] = { fg = c.fg },
-	["@text.literal.markdown_inline"] = { fg = c.purple },
+	["@label"] = { link = "Label" },
+	["@module"] = { link = "Structure" },
+	["@module.builtin"] = { fg = c.brown },
 
-	["@punctuation.special.markdown"] = { fg = c.brown },
+	-- Literals
+	["@string"] = { link = "String" },
```
Comment (confusedkernel) [inline lua/curry/theme.lua]: Is the empty line here intentional?

Comment (leana8959) [reply lua/curry/theme.lua]: Yeah I was trying to separate `@string` from the rest of `@string` related highlight groups, which are overrides

Comment (confusedkernel) [reply lua/curry/theme.lua]: Okay, seems good to me!

**Comment by leana8959** [comment]
@nottyl If you would like to, you can use this branch by switching to the PR branch `git fetch new_treesitter && git switch new_treesitter`.

Maybe we'll use this branch for a while to see if we encounter any weird color combination before un-drafting this PR ?

**Pull request merged**